api/v1/client.go

```diff
@@ -307,174 +238,55 @@ func (c *storageClient) Entities(ctx context.Context, r *http.Request) (*EntityL
 		return nil, common.ErrBadRequest
 	}
 
-	rows, err := c.db.Query(
-		ctx,
-		qf.EntitiesQuery(),
-		pagination.Limit,
-		pagination.Offset,
-	)
-	if err != nil {
-		c.logger.Info("query failed",
-			"request_id", ctx.Value(RequestIDContextKey),
-			"err", err.Error(),
-		)
-		return nil, common.ErrStorageError
-	}
-	defer rows.Close()
-
-	es := EntityList{
-		Entities: []Entity{},
-	}
-	for rows.Next() {
-		var e Entity
-		if err := rows.Scan(&e.ID, &e.Address); err != nil {
-			c.logger.Info("query failed",
-				"err", err.Error(),
-			)
-			return nil, common.ErrStorageError
-		}
-
-		es.Entities = append(es.Entities, e)
-	}
-
-	return &es, nil
+	return c.storage.Entities(ctx, &p)
 }
 
 // Entity returns a registered entity.
-func (c *storageClient) Entity(ctx context.Context, r *http.Request) (*Entity, error) {
-	cid, ok := ctx.Value(ChainIDContextKey).(string)
-	if !ok {
-		return nil, common.ErrBadChainID
-	}
-	qf := NewQueryFactory(cid)
+func (c *storageClient) Entity(ctx context.Context, r *http.Request) (*storage.Entity, error) {
+	var q storage.EntityRequest
 
 	entityID, err := url.PathUnescape(chi.URLParam(r, "entity_id"))
 	if err != nil {
 		return nil, common.ErrBadRequest
 	}
-
-	var e Entity
-	if err = c.db.QueryRow(
-		ctx,
-		qf.EntityQuery(),
-		entityID,
-	).Scan(&e.ID, &e.Address); err != nil {
-		c.logger.Info("row scan failed",
-			"request_id", ctx.Value(RequestIDContextKey),
-			"err", err.Error(),
-		)
-		return nil, common.ErrStorageError
-	}
-
-	entityID, err = url.PathUnescape(chi.URLParam(r, "entity_id"))
-	if err != nil {
+	if entityID == "" {
```

Comment:
I haven't found a suitable way to validate the [EntityID](https://pkg.go.dev/github.com/oasisprotocol/oasis-core/go/common/entity#Entity) yet. Pointers would be appreciated :)

Reply:
```
func validateEntityID(param string) (*signature.PublicKey, error) {
    var pk *signature.PublicKey
    if err := pk.UnmarshalHex(param); err != nil  || !pk.IsValid() {
        return nil, err
    }
    return pk, nil
}
```

is probably what you're looking for